### PR TITLE
Enable JSON-LD profiles for response serializations

### PR DESCRIPTION
**Pull request by awoods**
Resolves: https://jira.duraspace.org/browse/FCREPO-1615

### CODE REVIEW

**Comment by awoods** [comment]
Please do not squash these two commits.

File: fcrepo-http-commons/src/test/java/org/fcrepo/http/commons/responses/WriterConfigHelperTest.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import static org.openrdf.rio.helpers.JSONLDSettings.JSONLD_MODE;
+
+/**
+ * @author awoods
+ * @since 7/7/2015.
+ */
+@RunWith(Parameterized.class)
+public class WriterConfigHelperTest {
+
+    @Parameterized.Parameter(value = 0)
```
Comment (ajs6f) [inline fcrepo-http-commons/src/test/java/org/fcrepo/http/commons/responses/WriterConfigHelperTest.java]: Just for readability, please import `@Parameterized.Parameter` as `@Parameter`.

File: fcrepo-http-commons/src/test/java/org/fcrepo/http/commons/responses/WriterConfigHelperTest.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    public MediaType mediaType;
+
+    @Parameterized.Parameter(value = 1)
+    public JSONLDMode profile;
+
+    @Parameterized.Parameters(name = "{index}: mediaType:{0} => {1}")
```
Comment (ajs6f) [inline fcrepo-http-commons/src/test/java/org/fcrepo/http/commons/responses/WriterConfigHelperTest.java]: Just for readability, please import `@Parameterized.Parameters` as `@Parameters`.

File: pom.xml [hunk 3]
```diff
@@ -199,6 +199,25 @@
...
+        <exclusions>
+          <exclusion>
+            <groupId>log4j</groupId>
+            <artifactId>log4j</artifactId>
+          </exclusion>
+          <exclusion>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-log4j12</artifactId>
+          </exclusion>
+          <exclusion>
```
Comment (ajs6f) [inline pom.xml]: Is Jena pulling in an old version of `jsonld-java`?

File: fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/responses/RdfStreamStreamingOutput.java [hunk 4]
```diff
@@ -64,6 +64,8 @@
 
     private final RDFFormat format;
 
+    private final MediaType mediaType;
```
Comment (ajs6f) [inline fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/responses/RdfStreamStreamingOutput.java]: We may need a brief comment on these fields explaining how `format` and `mediaType` work together to develop a specific choice of serialization.

File: pom.xml (hunk 3 above)
Comment (awoods) [inline pom.xml]: Yes, there is a version conflict between the version of jsonld-java that jena pulls in and the most recent jsonld-java release.

Comment (ajs6f) [inline pom.xml]: Okay, cool.

**Pull request merged**